indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -698,6 +700,16 @@ private boolean announceTask(
     }
   }
 
+  private boolean cancelWorkerCleanup(String workerHost)
+  {
+    ScheduledFuture previousCleanup = removedWorkerCleanups.remove(workerHost);
+    if (previousCleanup != null) {
+      log.info("Cancelling Worker[%s] scheduled task cleanup", workerHost);
+      previousCleanup.cancel(false);
```

Comment:
It is possible yes. In that case it will just keep running. There's some chance of race between this code and the addWorker setup code (this code might delete a status path that addWorker just set a watch on)

What do you think about addressing that by having all this stuff (cleanup + addWorker + removeWorker) run in the same single threaded executor? I don't think it should cause too much of a problem since none of these tasks will take very long.